pattern_library/utils.py

```diff
@@ -197,12 +197,18 @@ def get_pattern_markdown(template_name):
         return markdown.markdown(f.read())
 
 
-def render_pattern(request, template_name, allow_non_patterns=False):
+def render_pattern(request, template_name, allow_non_patterns=False, config=None):
     if not allow_non_patterns and not is_pattern(template_name):
         raise TemplateIsNotPattern
 
-    context = get_pattern_context(template_name)
+    if not config:
+        config = get_pattern_config(template_name)
+
+    context = config.get('context', {})
+    tags = config.get('tags', {})
```

Comment:
I'm concerned that doing this here won't play nicely with any context modifiers that get run below. If the modifiers don't modify the tag config, or modify it in-place, then `tags` will contain the modified config. However, if any of the modifiers _replace_ `config['tags']` with another dict, then `tags` and `context['tags']` will refer to different objects, which we don't want!

I think the answer here is just to move these new lines down so they're after the context modifiers have run. What do you think?

Reply:
Are you sure this scenario is possible? context modifiers can only replace the context, not the whole config, unless I’m missing something.